ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/parameter/HateoasMapper.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.openapi.service.mapper.parameter;
+
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.openapi.service.mapper.hateoas.Resource;
+import io.ballerina.openapi.service.mapper.response.model.HateoasLink;
+import io.swagger.v3.oas.models.links.Link;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.openapi.service.mapper.Constants.OPENAPI_LINK_DEFAULT_REL;
+import static io.ballerina.openapi.service.mapper.hateoas.HateoasContextHolder.getHateoasContextHolder;
+import static io.ballerina.openapi.service.mapper.utils.MapperCommonUtils.getResourceConfigAnnotation;
+import static io.ballerina.openapi.service.mapper.utils.MapperCommonUtils.getValueForAnnotationFields;
+
+public class HateoasMapper {
```

Comment:
Is there any particular reason to create this Mapper under the parameter package? Shouldn't this come under the parent level at the same in `ServiceToOpenAPIMapper`

Reply:
Fixed with [63ac241d79644776e53dc2f338009c91970f6df7](https://github.com/ballerina-platform/openapi-tools/pull/1589/commits/63ac241d79644776e53dc2f338009c91970f6df7)